### DISCUSSION TITLE
[BUG] Block at XXX is no longer in the blockchain (it's in a fork)

### DISCUSSION THREAD

**Discussion by arturzx** [Chia-Network/chia-blockchain#4913] [Support]
**Describe the bug**
Exception in debug.log
`2021-05-14T04:47:33.642 full_node full_node_server        : ERROR    Exception: Block at e8c843c0659d43486c1df81d8b34f0f51af3be3d55e05cd13508863eb138635a is no longer in the blockchain (it's in a fork) <class 'ValueError'>, closing connection {'host': '186.220.36.43', 'port': 8444}. Traceback (most recent call last):
  File "/root/chia-blockchain/chia/server/server.py", line 531, in api_call
    response: Optional[Message] = await asyncio.wait_for(wrapped_coroutine(), timeout=timeout)
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 481, in wait_for
    return fut.result()
  File "/root/chia-blockchain/chia/server/server.py", line 528, in wrapped_coroutine
    raise e
  File "/root/chia-blockchain/chia/server/server.py", line 521, in wrapped_coroutine
    result = await coroutine
  File "/root/chia-blockchain/chia/full_node/full_node_api.py", line 244, in request_proof_of_weight
    wp = await self.full_node.weight_proof_handler.get_proof_of_weight(request.tip)
  File "/root/chia-blockchain/chia/full_node/weight_proof.py", line 77, in get_proof_of_weight
    wp = await self._create_proof_of_weight(tip)
  File "/root/chia-blockchain/chia/full_node/weight_proof.py", line 105, in _create_proof_of_weight
    recent_chain = await self._get_recent_chain(tip_rec.height)
  File "/root/chia-blockchain/chia/full_node/weight_proof.py", line 180, in _get_recent_chain
    headers = await self.blockchain.get_header_blocks_in_range(min_height, tip_height)
  File "/root/chia-blockchain/chia/consensus/blockchain.py", line 659, in get_header_blocks_in_range
    raise ValueError(f"Block at {block.header_hash} is no longer in the blockchain (it's in a fork)")
ValueError: Block at e8c843c0659d43486c1df81d8b34f0f51af3be3d55e05cd13508863eb138635a is no longer in the blockchain (it's in a fork)

2021-05-14T04:47:33.642 full_node full_node_server        : INFO     Connection closed: 186.220.36.43, node id: 8652cc8c0f5c5fc2e5b2b4ed3e33df67f1cab943756f8852b4bd150d1a307434`

**Comment by Plopsi** [comment]
hmm... and what is the problem? It's in an orphan and got dropped, pretty sure this is how it should work. Maybe only that this shouldn't log an error instead of an INFO

**Reply by sargonas** [reply]
Moving this to support discussions to archive it for posterity in case others search for a similar issue. As Plopsi stated above, this is expected behavior when a "bad" node is sending you info that it not in consensus with the rest of the chain you are part of. He brings up a good point that it might be better placed at the Info level and not Warning however, since you can't really control this inbound node connections coming into you, and they are being properly discarded as expected.

(For a little additional context, one assumes the reason why you had this happen is this is a node that has not updated to 1.1.5 since the issue that occurred on Sunday, 5/9, and is presumably part of a forked group of nodes still exchanging bad data until they are patched.)

**Reply by arturzx** [reply]
Thank you for explantation.

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
